Rubberduck.Core/AutoComplete/AutoCompleteKeyDownHandler.cs

```diff
@@ -0,0 +1,209 @@
+                    }
+                }
+            }
+        }
+
+        private CodeString GetEntireLogicalCodeLine(ICodeModule module, Selection pSelection)
+        {
+            var lines = new List<(int Line, string Content)>();
+            var currentLineIndex = pSelection.StartLine;
+            var currentLine = module.GetLines(currentLineIndex, 1);
+
+            var caretLine = (currentLineIndex, currentLine);
+            lines.Add(caretLine);
+
+            while (currentLineIndex >= 1)
+            {
+                currentLineIndex--;
+                if (currentLineIndex >= 1)
+                {
+                    currentLine = module.GetLines(currentLineIndex, 1);
+                    if (currentLine.Replace("\r\n", string.Empty).EndsWith(" _"))
+                    {
+                        lines.Insert(0, (currentLineIndex, currentLine));
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+
+            currentLineIndex = pSelection.StartLine;
+            currentLine = caretLine.currentLine;
+            while (currentLineIndex <= module.CountOfLines && currentLine.Replace("\r\n", string.Empty).EndsWith(" _"))
+            {
+                currentLineIndex++;
+                if (currentLineIndex <= module.CountOfLines)
+                {
+                    currentLine = module.GetLines(currentLineIndex, 1);
+                    lines.Add((currentLineIndex,currentLine));
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            var logicalLine = string.Join("\r\n", lines.Select(e => e.Content));
+            var zCaretLine = lines.IndexOf(caretLine);
+            var zCaretColumn = pSelection.StartColumn - 1;
+
+            var startLine = lines[0].Line;
+            var endLine = lines[lines.Count - 1].Line;
+
+            var result = new CodeString(logicalLine, new Selection(zCaretLine, zCaretColumn), new Selection(startLine, 1, endLine, lines[lines.Count - 1].Content.Length));
+            return result;
+        }
+
+        private bool IsInsideStringLiteral(Selection pSelection, ref string currentContent)
+        {
+            if (!currentContent.Substring(pSelection.StartColumn - 1).Contains("\"") ||
+                currentContent.StripStringLiterals().HasComment(out _))
+            {
+                return false;
+            }
+
+            var zSelection = pSelection.ToZeroBased();
+            var leftOfCaret = currentContent.Substring(0, zSelection.StartColumn);
+            var rightOfCaret = currentContent.Substring(Math.Min(zSelection.StartColumn + 1, currentContent.Length - 1));
+            if (!rightOfCaret.Contains("\""))
+            {
+                // the string isn't terminated, but VBE would terminate it here.
+                currentContent += "\"";
+                rightOfCaret += "\"";
```

Comment:
@Vogel612 because if the caret *isn't* inside a string literal, then we won't be interfering with the code pane in any way - basically we defer "prettifying" as much as possible, and only when we absolutely need to.